fitbenchmarking/cli/checkpoint_handler.py

```diff
@@ -110,6 +136,205 @@ def generate_report(options_file='', additional_options=None, debug=False):
+             costfun_tag they are considered identical.
+         4b) If A an B share identical results, the details not specified in 4a
+             are taken from A if strategy is 'first', or B if strategy is
+             'last'.
+     5) As tables are grids of results, combining arbitrary results can lead to
+        un-table-able checkpoint files.
+        This occurs when the problems in A and B are not all identical and the
+        set of combinations of software_tag, minimizer_tag, jacobian_tag,
+        hessian_tag, and costfun_tag for which there are results in each of A
+        and B are not identical.
+        E.g. B has a problem not in A and uses a minimizer for which there are
+             no results in A.
+         5a) If the resulting checkpoint file would have the above issue, the
+             checkpoints are incompatible.
+         5b) Incompatible checkpoint files can be combined but should raise
+             warnings and mark the dataset in the checkpoint file.
+             Note: Some datasets may be incompatible where others can be
+                   successfully combined.
+     6) Unselected minimizers and failed problems will be discarded when
+        combining.
+
+    :param files: The files to combine.
+    :type files: list[str]
+    :param output: The name for the new checkpoint file.
+    :type output: str
+    :param strategy: The stategy for merging identical elements.
+                     Options are 'first' or 'last'.
+    :type strategy: str
+    :param debug: Enable debugging output.
+    :type debug: bool
+    """
+    if len(files) < 2:
+        return
+    if strategy == 'first':
+        pass
+    elif strategy == 'last':
+        files = list(reversed(files))
+    print(f"Loading {files[0]}...")
+    with open(files[0], 'r', encoding='utf-8') as f:
+        A = json.load(f)
+    for to_merge in files[1:]:
+        print(f"Merging {to_merge}...")
+        with open(to_merge, 'r', encoding='utf-8') as f:
+            B = json.load(f)
+        A = merge(A, B)
+
+    print(f'Writing to {output}...')
+    with open(output, 'w', encoding='utf-8') as f:
+        json.dump(A, f, indent=2)
+
+
+def merge(A, B):
+    """
+    Merge the results from A and B
+    This function can corrupt A and B, they should be discarded after calling.
+
+    :param A: The first set of results to merge
+    :type A: dict[str, list[FittingResult]]
+    :param B: The set to merge into A
+    :type B: dict[str, list[FittingResult]]
+
+    :return: The merged checkpoint data.
+    :rtype: dict[str, any]
+    """
+    for k in B:
+        if k not in A:
+            A[k] = B[k]
+            continue
+        A[k]['problems'], update_names = merge_problems(A[k]['problems'],
+                                                        B[k]['problems'])
+        for r in B[k]['results']:
+            if r['name'] in update_names:
+                r['name'] = update_names[r['name']]
```

Comment:
```suggestion
        if update_names:
            for r in B[k]['results']:
                if r['name'] in update_names:
                    r['name'] = update_names[r['name']]
```

Maybe we can add the `if update_names` outside this loop. I was testing it with A.json and B.json and found that the loop was running even when updates_names was empty.